### PR TITLE
docs: extend documentation

### LINKED ISSUE #89: Update TRS-Filer Documentation
Add details about the basic flow, implementation, and usage of TRS-Filer.

### PR DESCRIPTION
**Pull request by kushagra189**
## Description

Update `README.md` with description, usage and implementation details

Fixes #89 

## Type of change

Please delete options that are not relevant.

- [x] Documentation update

## Checklist:

- [x] My code follows the [style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/contributing_guidelines.md#language-specific-guidelines) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [x] I have added docstrings following the [Python style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/python.md) of this project to all new modules, classes, methods and functions are documented with docstrings following; I have updated any previously existing docstrings, if applicable
- [x] I have updated any sections of the app's documentation that are affected by the proposed changes, if applicable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,212 @@
...
+in [`additions.openapi.yaml`][trs-filer-api-custom], the application would
+expect the corresponding controller to be implemented in function/class
+`ga4gh.trs.server.myController()`.
+
+> If necessary, the default location of the module containing the controllers
+> can be customized following instructions provided by [FOCA][res-foca],
+> [Connexion][res-connexion] and [OpenAPI][res-openapi].
+
+Custom errors can be specified in module
+[`errors.exceptions`][trs-filer-exceptions]. Just follow the existing structure
```
Comment (uniqueg) [inline README.md:198]: Link to `errors.exceptions` is broken, link target needs to contain `trs_filer/` in path

**Review by uniqueg — CHANGES_REQUESTED** [review]
Looks great - thanks a lot! Please fix the one broken link

**Review by uniqueg — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -36,21 +222,48 @@ This project is covered by the [Apache License 2.0][license-apache] also
...
-[badge-coverage]:<https://img.shields.io/coveralls/github/elixir-cloud-aai/trs-filer>
-[badge-github-tag]:<https://img.shields.io/github/v/tag/elixir-cloud-aai/trs-filer?color=C39BD3>
+[badge-build-status]:<https://travis-ci.com/elixir-cloud-aai/cloud-registry.svg?branch=dev>
+[badge-coverage]:<https://img.shields.io/coveralls/github/elixir-cloud-aai/cloud-registry>
 [badge-license]:<https://img.shields.io/badge/license-Apache%202.0-blue.svg>
-[badge-url-build-status]:<https://travis-ci.com/elixir-cloud-aai/trs-filer>
-[badge-url-coverage]:<https://coveralls.io/github/elixir-cloud-aai/trs-filer>
-[badge-url-github-tag]:<https://github.com/elixir-cloud-aai/trs-filer/releases>
+[badge-url-build-status]:<https://travis-ci.com/elixir-cloud-aai/cloud-registry>
+[badge-url-coverage]:<https://coveralls.io/github/elixir-cloud-aai/cloud-registry>
```
Comment (sarthakgupta072) [inline README.md:236]: @kushagra189 these links should point to `trs-filer`, but are pointing to `cloud-registry`.

Comment (kushagra189) [reply README.md:236]: Will check and fix by today.

**Pull request merged**